### PR TITLE
Lost rowset during tablet revise tablet meta

### PR DESCRIPTION
**Pull request by yiguolei**
#1966

### CODE REVIEW

**Review by chaoyli — DISMISSED** [review]

File: be/src/olap/tablet.h [hunk 1]
```diff
@@ -242,6 +242,8 @@ class Tablet : public std::enable_shared_from_this<Tablet> {
 
     void build_tablet_report_info(TTabletInfo* tablet_info);
 
+    OLAPStatus clone_tablet_meta(TabletMetaSharedPtr new_tablet_meta);
```
Comment (imay) [inline be/src/olap/tablet.h]: What locks does caller need to hold? one or two?
And why not implement as a TabletMeta static function? which like
static copy(const TabletMeta& src, TabletMeta* dest)?

If the function name is clone, I don't know the direction of clone operation.

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/olap/tablet.h (hunk 1 above)
Comment (imay) [reply be/src/olap/tablet.h]: And you should comment for this function why we need it other than get it from RocksDB

Comment (yiguolei) [reply be/src/olap/tablet.h]: Need tablet header lock.
Could not implement as a static method because need tablet header lock.

Comment (imay) [reply be/src/olap/tablet.h]: In your implementation, there is no lock.

Comment (yiguolei) [reply be/src/olap/tablet.h]: Callers acquire the header lock.

Comment (yiguolei) [reply be/src/olap/tablet.h]: In our implementation, many methods in tablet not have header lock but the callers get tablet header lock and then call the method.

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**